src/rebar_prv_eunit.erl

```diff
@@ -134,17 +138,28 @@ resolve_tests(State) ->
     Files        = resolve(file, RawOpts),
     Modules      = resolve(module, RawOpts),
     Suites       = resolve(suite, module, RawOpts),
-    Apps ++ Applications ++ Dirs ++ Files ++ Modules ++ Suites.
+    Generator    = resolve(generator, RawOpts),
+    Apps ++ Applications ++ Dirs ++ Files ++ Modules ++ Suites ++ Generator.
 
 resolve(Flag, RawOpts) -> resolve(Flag, Flag, RawOpts).
 
 resolve(Flag, EUnitKey, RawOpts) ->
     case proplists:get_value(Flag, RawOpts) of
         undefined -> [];
-        Args      -> lists:map(fun(Arg) -> normalize(EUnitKey, Arg) end,
-                               rebar_string:lexemes(Args, [$,]))
+        Args      ->
+            lists:flatten(lists:map(fun(Arg) -> normalize(EUnitKey, Arg) end,
+                                    rebar_string:lexemes(Args, [$,])))
     end.
 
+normalize(generator, Value) ->
+    case string:tokens(Value, [$:]) of
+        [Module0, Functions] ->
+            Module = list_to_atom(Module0),
+            lists:map(fun(F) -> {generator, Module, list_to_atom(F)} end,
+                      string:tokens(Functions, [$;]));
+        _ ->
+            throw(lists:concat(["Generator `", Value, "' is invalid format."]))
```

Comment:
I'd expect the exception value to be wrapped with `?PRV_ERROR({generator, Val})` in order for rebar3 to call back `format_error({generator, Val})` to get the final string value. This allows you to use `{generator, Val}` as the return value in API mode while having the proper string output to the user.